src/map/lua/lua_baseentity.cpp

```diff
@@ -10818,42 +10818,29 @@ void CLuaBaseEntity::delLearnedAbility(uint16 abilityID)
  *  Notes   : Optional config table accepts bools for: silentLog, saveToDB, sendUpdate.
  ************************************************************************/
 
-void CLuaBaseEntity::addSpell(uint16 spellID, sol::object const& arg0)
+void CLuaBaseEntity::addSpell(uint16 spellID, sol::optional<sol::table> const& paramTable)
 {
     if (m_PBaseEntity->objtype != TYPE_PC)
     {
         ShowWarning("Invalid entity type calling function (%s).", m_PBaseEntity->getName());
         return;
     }
 
-    auto* PChar = static_cast<CCharEntity*>(m_PBaseEntity);
-
-    bool silentLog  = false;
-    bool saveToDB   = true;
-    bool sendUpdate = true;
-
-    if (arg0.get_type() == sol::type::table)
-    {
-        for (const auto& kv : arg0.as<sol::table>())
-        {
-            if (kv.first.get_type() == sol::type::string && kv.second.get_type() == sol::type::boolean)
-            {
-                auto keyName = kv.first.as<std::string>();
-                if (keyName == "silentLog")
-                {
-                    silentLog = kv.second.as<bool>();
-                }
-                else if (keyName == "saveToDB")
-                {
-                    saveToDB = kv.second.as<bool>();
-                }
-                else if (keyName == "sendUpdate")
-                {
-                    sendUpdate = kv.second.as<bool>();
-                }
+    auto coalesceParam = [](const sol::optional<sol::table>& maybeParamTable, const std::string& key, bool defaultValue) -> bool {
+        if (maybeParamTable && maybeParamTable->valid()) {
+            sol::table t = *maybeParamTable;
```

Comment:
Sol claims `sol::table` is extremely efficient, but I don't really know the guts here: https://sol2.readthedocs.io/en/latest/api/table.html

Test this out and see if we can instead use this pattern: `const auto& thing = *maybeThing;`